google/cloud/internal/log_wrapper.cc

```diff
@@ -32,6 +34,32 @@ std::string DebugString(google::protobuf::Message const& m,
   return str;
 }
 
+std::string RequestIdForLogging() {
+  thread_local std::uint64_t generator = 0;
+  std::ostringstream os;
+  os << std::this_thread::get_id() << ":" << ++generator;
```

Comment:
I don't think this is guaranteed to be unique.  When a thread ends, objects with thread storage duration are destroyed, and the thread ID is available for reuse.  So, you could see something like "X:1" twice.  Perhaps a static atomic is better.

Reply:
Good point, fixed.

Reply:
`int` can be just 32 bits, I think.  If so, a binary with 1Kqps would overflow in less than a month.

I don't know if that rate or uptime is realistic, but I'd probably burn an extra 32 bits to avoid the issue.

In either case, I'd also suggest an unsigned type so that overflow is at least defined.